### ISSUE TITLE
S3 Dynamic Bucket Creation - Dataset Feature

### ISSUE THREAD

**Issue by Eternity022** [datashim-io/datashim#301]
Hello all,

I was just wondering, is the Dataset provides the option to dynamically create the bucket, if the bucket doesn't already exist?

If it's not, is there any plan of adding this feature?

Thanks :)

**Comment by srikumar003** [comment]
@Eternity022 We do have the feature to create the bucket ! Here is [the relevant example](https://github.com/datashim-io/datashim/blob/8ec79a3aa25334c287dd2c877a2e3765c441f6ca/examples/templates/example-dataset-s3-provision.yaml)

**Comment by Eternity022** [comment]
So this manifest will create the bucket, if the bucket doesn't exist in the backend Object Store?

**Comment by srikumar003** [comment]
yes, `provision: true` intitiates creation of the bucket. Typo: in the example above, we have set `readonly: true` when it should be `false` in practice

**Comment by Eternity022** [comment]
Thank you for your response. We will start testing :) You can archive this Question if you want, or leave it for the community. :)
